shared/chat/search.js

```diff
@@ -0,0 +1,123 @@
+// @flow
+import React from 'react'
+import * as Creators from '../actions/chat/creators'
+import * as SearchCreators from '../actions/searchv3/creators'
+import {getProfile} from '../actions/tracker'
+import * as SearchConstants from '../constants/searchv3'
+import UserInput from '../searchv3/user-input'
+import SearchResultsList from '../searchv3/results-list'
+import ServiceFilter from '../searchv3/services-filter'
+import {Box, Icon} from '../common-adapters'
+import {compose, withState, defaultProps, withHandlers} from 'recompose'
+import {connect} from 'react-redux'
+import {globalStyles, globalMargins} from '../styles'
+import {parseUserId, serviceIdToIcon} from '../util/platforms'
+import {chatSearchResultArray} from '../constants/selectors'
+import {onChangeSelectedSearchResultHoc} from '../searchv3/helpers'
+
+import type {TypedState} from '../constants/reducer'
+
+const mapStateToProps = (state: TypedState) => {
```

Comment:
we should use recompose here else this'll always fail the connect cache and always rerender right